staging/src/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go

```diff
@@ -122,7 +122,11 @@ func (m *rootScopedMetrics) getForNamespace(namespace string, metricName string,
 		return nil, err
 	}
 
-	res := metricObj.(*v1beta2.MetricValueList)
+	var res *v1beta2.MetricValueList
+	var ok bool
+	if res, ok = metricObj.(*v1beta2.MetricValueList); !ok {
+		return nil, fmt.Errorf("the custom metrics API server didn't return MetricValueList")
```

Comment:
Can you check if the metricObj is of type *v1.Status and provide appropriate error code?

Reply:
Can you use @ to specify who is meant with **you**, thanks

Reply:
@tedyu 
I meant the author of the PR

Reply:
What appropriate error code should be returned ?
Something like this ?
metricObj is of type *v1.Status

Reply:
@kawych 
Were you thinking about something like this ?
```
	if res, ok = metricObj.(*v1beta2.MetricValueList); !ok {
		if _, ok = metricObj.(*v1.Status); ok {
			return nil, fmt.Errorf("the custom metrics API server returned v1.Status instead of MetricValueList")
		}
		return nil, fmt.Errorf("the custom metrics API server didn't return MetricValueList")
	}
```

Reply:
I have included the type of metricObj in the error message.